peps/pep-0764.rst

```diff
@@ -0,0 +1,300 @@
+PEP: 764
+Title: Inlined typed dictionaries
+Author: Victorien Plot <contact@vctrn.dev>
+Sponsor: Eric Traut <erictr at microsoft.com>
+Status: Draft
+Type: Standards Track
+Topic: Typing
+Created: 25-Oct-2024
+Python-Version: 3.14
+
+.. highlight:: python
+
+
+Abstract
+========
+
+:pep:`589` defines a :ref:`class-based <typing:typeddict-class-based-syntax>`
+and a :ref:`functional syntax <typing:typeddict-functional-syntax>` to create
+typed dictionaries. In both scenarios, it requires defining a class or
+assigning to a value. In some situations, this can add unnecessary boilerplate,
+especially if the typed dictionary is only used once.
+
+This PEP proposes the addition of a new inlined syntax, by subscripting the
+:class:`~typing.TypedDict` type::
+
+    from typing import TypedDict
+
+    def get_movie() -> TypedDict[{'name': str, 'year': int}]:
+        return {
+            'name': 'Blade Runner',
+            'year': 1982,
+        }
+
+Motivation
+==========
+
+Python dictionaries are an essential data structure of the language. Many
+times, it is used to return or accept structured data in functions. However,
+it can get tedious to define :class:`~typing.TypedDict` classes:
+
+* A typed dictionary requires a name, which might not be relevant.
+* Nested dictionaries requires more than one class definition.
+
+Taking a simple function returning some nested structured data as an example::
+
+    from typing import TypedDict
+
+    class ProductionCompany(TypedDict):
+        name: str
+        location: str
+
+    class Movie(TypedDict):
+        name: str
+        year: int
+        production: ProductionCompany
+
+
+    def get_movie() -> Movie:
+        return {
+            'name': 'Blade Runner',
+            'year': 1982,
+            'production': {
+                'name': 'Warner Bros.',
+                'location': 'California',
+            }
+        }
+
+
+Rationale
+=========
+
+The new inlined syntax can be used to resolve these problems::
+
+    def get_movie() -> TypedDict[{'name': str, 'year': int, 'production': TypedDict[{'name': str, 'location': str}]}]:
+        ...
+
+It is recommended to *only* make use of inlined typed dictionaries when the
+structured data isn't too large, as this can quickly become hard to read.
+
+While less useful (as the functional or even the class-based syntax can be
+used), inlined typed dictionaries can be assigned to a variable, as an alias::
+
+    InlinedTD = TypedDict[{'name': str}]
```

Comment:
Perhaps a bit more explanation here could be helpful, showing an example of where this `InlinedTD` could be practically used?

Also, the wording here is a bit confusing to me. Maybe we could switch up the order:

```
Inlined typed dictionaries can be assigned to a variable, as an alias:

InlinedTD = TypedDict[{'name': str}]

< insert example of where this is helpful>

< note about this potentially being less useful bc of the utility of functional / class based syntax for this purpose>
```